kraken-build/src/kraken/std/python/tasks/publish_task.py

```diff
@@ -33,31 +33,28 @@ def get_relationships(self) -> Iterable[TaskRelationship]:
 
     def execute(self) -> TaskStatus:
         credentials = self.index_credentials.get()
-        repository_url = self.index_upload_url.get().rstrip("/") + "/"
         command = [
-            *self.twine_cmd.get(),
-            "upload",
-            "--repository-url",
-            repository_url,
-            "--verbose",
+            "uv",
+            "publish",
+            "--publish-url",
+            self.index_upload_url.get(),
             *[str(x.absolute()) for x in self.distributions.get()],
         ]
-        if credentials:
-            command += [
-                "--username",
-                credentials[0],
-                "--password",
-                credentials[1],
-            ]
-        if not self.interactive.get():
-            command.append("--non-interactive")
         if self.skip_existing.get():
-            command.append("--skip-existing")
+            command.extend(["--check-url", self.index_check_url.get() or self.index_index_url.get()])
+
+        env = os.environ.copy()
+        if credentials:
+            # See https://docs.astral.sh/uv/guides/package/#publishing-your-package
+            # NOTE: PyPI does not support publishing with username and password anymore,
+            #       should we log a warning if the username is not __token__?
+            env["UV_PUBLISH_USERNAME"] = credentials[0]
+            env["UV_PUBLISH_PASSWORD"] = credentials[1]
 
-        safe_command = [x.replace(credentials[1], "MASKED") for x in command] if credentials else command
+        safe_command = command
         self.logger.info("$ %s", safe_command)
 
-        returncode = subprocess.call(command, cwd=self.project.directory)
+        returncode = subprocess.call(command, cwd=self.project.directory, env=env)
         return TaskStatus.from_exit_code(safe_command, returncode)
```

Comment:
```suggestion
    try:
        returncode = subprocess.call(command, cwd=self.project.directory, env=env)
    except FileNotFoundError:
        return TaskStatus.failed(
            "The 'uv' command was not found. The python.publish task requires uv to be installed. "
            "See https://github.com/astral-sh/uv for installation instructions."
        )

        return TaskStatus.from_exit_code(safe_command, returncode)
```

Reply:
If we use the `uv` binary that comes with Kraken through it's dependencies, we wouldn't need that check. That UV version is pinned with `.kraken.lock`, which seems like a desirable property; although on the other hand it might mean the `uv` version run by users when invoked manually is likely a different one..